packages/perspective-viewer-d3fc/src/js/charts/sunburst.js

```diff
@@ -0,0 +1,141 @@
+ *
+ * Copyright (c) 2017, the Perspective Authors.
+ *
+ * This file is part of the Perspective library, distributed under the terms of
+ * the Apache License 2.0.  The full license can be found in the LICENSE file.
+ *
+ */
+
+import {select} from "d3";
+import {treeData} from "../data/treeData";
+import {clickHandler} from "../interaction/clickHandler";
+import {drawArc, arcVisible} from "../series/arcSeries";
+import {labelVisible, labelTransform, cropLabel} from "../axis/sunburstLabel";
+import {colorRangeLegend} from "../legend/colorRangeLegend";
+import {tooltip} from "../tooltip/tooltip";
+
+function sunburst(container, settings) {
+    const sunburstData = treeData(settings);
+    const {width: containerWidth, height: containerHeight} = container.node().getBoundingClientRect();
+
+    const minSize = 500;
+    const cols = sunburstData.length === 1 ? 1 : Math.floor(containerWidth / minSize);
+    const rows = Math.ceil(sunburstData.length / cols);
+    container.style("grid-template-columns", `repeat(${cols}, ${containerWidth / cols}px)`);
+    container.style("grid-template-rows", `repeat(${rows}, ${containerHeight / cols}px)`);
+
+    const sunburstDiv = container.selectAll("div.sunburst-container").data(treeData(settings), d => d.split);
+    sunburstDiv.exit().remove();
+
+    const sunburstEnter = sunburstDiv
+        .enter()
+        .append("div")
+        .attr("class", "sunburst-container");
+
+    const sunburstContainer = sunburstEnter
+        .append("svg")
+        .append("g")
+        .attr("class", "sunburst");
+
+    sunburstContainer.append("text").attr("class", "title");
+
+    sunburstContainer
+        .append("circle")
+        .attr("fill", "none")
+        .attr("pointer-events", "all");
+
+    sunburstContainer.append("text").attr("class", "parent");
+    sunburstEnter
+        .merge(sunburstDiv)
+        .select("svg")
+        .select("g.sunburst")
+        .attr("transform", `translate(${containerWidth / 2 / cols}, ${containerHeight / 2 / cols})`)
+        .each(function({split, data, color}) {
+            const sunburstElement = select(this);
+            const svgNode = this.parentNode;
+            const {width, height} = svgNode.getBoundingClientRect();
+
+            const title = sunburstElement.select("text.title").text(split);
+            title.attr("transform", `translate(0, ${-(height / 2 - 5)})`);
+
+            const radius = (Math.min(width, height) - 100) / 6;
+            data.each(d => {
+                d.current = d;
+                d.mainValues = settings.mainValues.length === 1 ? d.value : [d.value, d.data.color];
+                d.crossValue = d
+                    .ancestors()
+                    .slice(0, -1)
+                    .reverse()
+                    .map(cross => cross.data.name)
+                    .join("|");
+                d.key = split;
+            });
+
+            const segment = sunburstElement.selectAll("g.segment").data(data.descendants().slice(1));
```

Comment:
Once we get to here, it seems like `segment` (and all its related stuff below) is analogous to the `series` in other charts (e.g. `lineSeries` or `barSeries`). Perhaps we can reduce the amount of code in the chart module by moving the series-related stuff into `/series/arcSeries.js` instead?